geode-core/src/main/java/org/apache/geode/internal/cache/DiskStoreImpl.java

```diff
@@ -2318,7 +2318,7 @@ void close(boolean destroy) {
       } catch (RuntimeException e) {
         rte = e;
       }
-      if (!isOffline()) {
```

Comment:
Wouldn't it be better to change the `DiskStoreImpl` constructor to only instantiate the `AsyncFlusherThread` whenever we are not offline instead?, that would address the issue and make the class consistent, right?.